### PR TITLE
Use Environment.TickCount instead of DateTime.UtcNow when waiting for some task

### PR DESCRIPTION
**Pull request by kagikn**
## Summary
- Improve performance when scripts wait for the signals or when some task (e.g. loading some model) waits for something or the wait timeout expires at the cost of the resolution of the timer. It will be limited to in the range of 10 milliseconds to 16 milliseconds, but GTA V itself causes major issues when running at more than 187 FPS, so the new resolution shouldn't be a major issue too much.
- No longer get influcenced by clock adjustment by the user

### CODE REVIEW

**Comment by kagikn** [comment]
Why I made this PR instead of directly mading commits? Because I wonder there are better designs instead of exposing `GetTickCount64` via `WinApiWrapper`. If there was `Environment.TickCount64` in .NET Framework, I'd directly made commits for the change.

**Comment by crosire** [comment]
`Environment.Tick` does exist in .NET Framework, which IMHO would be cleaner to use. Just has the nasty detail of returning a signed 32-bit value, which can wrap around after just 25 days of uptime, so may have to add an extra check to ensure an overflow won't trigger an infinite loop. E.g.:
```c#
try
{
	checked
	{
		int resumeTime = Environment.TickCount + ms;

		do
		{
			waitEvent.Release();
			continueEvent.Wait();
		}
		while (Environment.TickCount < resumeTime);
	}
}
catch (OverflowException)
{
	// Tick count overflowed, can't wait until it settled again =(
}
```

**Comment by kagikn** [comment]
I think we have a workaround without calling `GetTickCount64`. Calculate how much tick is elapsed since we don't expect the elapsed time to exceed more than about 25 days.

**Comment by crosire** [comment]
Ah right, that works. But IMHO this doesn't warrant a new class and file added, can just do this inline in all the places.

File: ScriptHookVDotNet.vcxproj [hunk 1]
```diff
@@ -138,35 +138,11 @@
...
-      TreatWarningsAsErrors="%(CsCompile.TreatWarningsAsErrors)"
-      Utf8Output="%(CsCompile.Utf8Output)"
-      WarningLevel="%(CsCompile.WarningLevel)"
-      WarningsAsErrors="%(CsCompile.WarningsAsErrors)"
-      WarningsNotAsErrors="%(CsCompile.WarningsNotAsErrors)"
-      ToolExe="$(CscToolExe)"
-      ToolPath="$(CscToolPath)" />
+  <Target Name="CsCompile" BeforeTargets="ClCompile" Inputs="@(CsCompile)" Outputs="$(IntDir)$(TargetName).netmodule">
+    <Csc Sources="@(CsCompile)" AllowUnsafeBlocks="%(CsCompile.AllowUnsafeBlocks)" DebugType="%(CsCompile.DebugType)" DisabledWarnings="%(CsCompile.NoWarn)" DocumentationFile="$(IntDir)$(TargetName).xdc" EmitDebugInformation="%(CsCompile.DebugSymbols)" LangVersion="%(CsCompile.LangVersion)" Optimize="%(CsCompile.Optimize)" OutputAssembly="$(IntDir)$(TargetName).netmodule" PdbFile="$(IntDir)$(TargetName).netmodule.pdb" TargetType="Module" TreatWarningsAsErrors="%(CsCompile.TreatWarningsAsErrors)" Utf8Output="%(CsCompile.Utf8Output)" WarningLevel="%(CsCompile.WarningLevel)" WarningsAsErrors="%(CsCompile.WarningsAsErrors)" WarningsNotAsErrors="%(CsCompile.WarningsNotAsErrors)" ToolExe="$(CscToolExe)" ToolPath="$(CscToolPath)" />
```
Comment (crosire) [inline ScriptHookVDotNet.vcxproj:142]: Please just revert this formatting change in the project file (which Visual Studio applied when you added the new file), and this is good to go.

File: source/scripting_v3/GTA/Weapons/WeaponAsset.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 //
 
 using GTA.Native;
+using SHVDN;
```
Comment (crosire) [inline source/scripting_v3/GTA/Weapons/WeaponAsset.cs:7]: Oh, this line can be reverted as well, now that there is no helper class used.

**Pull request merged**